### DISCUSSION TITLE
When run apply_to_state_machine or append_to_log failed. How should to do

### DISCUSSION THREAD

**Discussion by yanyun827** [databendlabs/openraft#909] [Q&A]
When an error occurs in those callback, openraft will exit the main loop, causing the program not run correctly; How to solve this situation.

**Comment by drmingdrmer** [comment]
If it is a storage error, e.g., data corruption, it should return an "StorageError" to inform Openraft to shutdown at once.

If it's an application error, the error should be wrapped in the `Response`, i.e., in this case, `Response` should be a `Result<T, E>`.

```rust
async fn apply_to_state_machine(
        &mut self,
        entries: &[Entry<TypeConfig>],
    ) -> Result<Vec<Response>, StorageError<NodeId>>
```
